Modules/Sources/Yosemite/Tools/POS/POSCart.swift

```diff
@@ -67,6 +116,96 @@ extension [POSCartItem] {
         return true
     }
 
+    func compareWithOrder(_ order: Order?) -> ItemsComparison {
+        guard let order else {
+            // If there's no order but we have items, all items are missing
+            let missingItems = self.compactMap { cartItem -> CartOrderComparison.MissingCartItem? in
+                guard let (productID, variationID) = extractProductIDs(from: cartItem.item) else {
+                    return nil
+                }
+                return CartOrderComparison.MissingCartItem(
+                    productID: productID,
+                    variationID: variationID,
+                    name: cartItem.item.name
+                )
+            }
+            return ItemsComparison(missingItems: missingItems, quantityMismatches: [])
+        }
+
+        // Group cart items by product/variation ID to consolidate duplicates
+        let cartItemsByProductKey = Dictionary(grouping: self, by: { item -> String in
+            guard let (productID, variationID) = extractProductIDs(from: item.item) else {
+                return "invalid_\(UUID())"
```

Comment:
I was confused by this UUID here, I believe it refers to the change to ensure that only the specific variation that failed is removed from the cart? If that so I'd add a comment for future reference.